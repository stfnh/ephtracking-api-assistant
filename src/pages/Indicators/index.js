import React, { Component, Fragment } from 'react';

import CodeBlock from '../../components/CodeBlock';
import Preview from '../../components/Preview';
import SelectContentArea from '../../containers/SelectContentArea';

class Indicators extends Component {
  constructor(props) {
    super(props);
    this.state = {
      contentAreaId: null
    };
    this.setContentAreaId = this.setContentAreaId.bind(this);
  }

  setContentAreaId(contentAreaId) {
    this.setState({ contentAreaId });
  }

  render() {
    const { contentAreaId } = this.state;
    const validUrl = contentAreaId !== null;
    const url = `https://ephtracking.cdc.gov/apigateway/api/v1/indicators/${contentAreaId}`;
    return (
      <Fragment>
        <h1 className="title">Retrieving List of all Content Areas</h1>
        <h5 className="title is-5">Usage</h5>
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/{'{'}version{'}'}/indicators/{
            '{'
          }contentAreaId{'}'}[?apiToken]
        </CodeBlock>
        <hr />
        <h5 className="title is-5">Set parameters</h5>
        <SelectContentArea handleSelect={this.setContentAreaId} />
        <hr />
        <Preview url={url} validUrl={validUrl} />
      </Fragment>
    );
  }
}

export default Indicators;
