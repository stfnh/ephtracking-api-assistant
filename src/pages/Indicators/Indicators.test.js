import React from 'react';
import { shallow } from 'enzyme';

import Indicators from './';

describe('Page Indicators', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Indicators />);
    expect(wrapper).toMatchSnapshot();
  });

  it('setsContentAreaId correctly', () => {
    const wrapper = shallow(<Indicators />);
    wrapper.instance().setContentAreaId('123');
    expect(wrapper.state().contentAreaId).toBe('123');
  });
});
