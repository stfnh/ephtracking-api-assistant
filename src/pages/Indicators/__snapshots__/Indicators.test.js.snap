// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Indicators renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Indicators />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Retrieving List of all Content Areas
</h1>,
        <h5
          className="title is-5"
>
          Usage
</h5>,
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/
          {
          version
          }
          /indicators/
          {
          contentAreaId
          }
          [?apiToken]
</CodeBlock>,
        <hr />,
        <h5
          className="title is-5"
>
          Set parameters
</h5>,
        <SelectContentArea
          handleSelect={[Function]}
/>,
        <hr />,
        <Preview
          url="https://ephtracking.cdc.gov/apigateway/api/v1/indicators/null"
          validUrl={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retrieving List of all Content Areas",
          "className": "title",
        },
        "ref": null,
        "rendered": "Retrieving List of all Content Areas",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Usage",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Usage",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/indicators/",
            "{",
            "contentAreaId",
            "}",
            "[?apiToken]",
          ],
        },
        "ref": null,
        "rendered": Array [
          "https://ephtracking.cdc.gov/apigateway/api/",
          "{",
          "version",
          "}",
          "/indicators/",
          "{",
          "contentAreaId",
          "}",
          "[?apiToken]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Set parameters",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Set parameters",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSelect": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/indicators/null",
          "validUrl": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Retrieving List of all Content Areas
</h1>,
          <h5
            className="title is-5"
>
            Usage
</h5>,
          <CodeBlock>
            https://ephtracking.cdc.gov/apigateway/api/
            {
            version
            }
            /indicators/
            {
            contentAreaId
            }
            [?apiToken]
</CodeBlock>,
          <hr />,
          <h5
            className="title is-5"
>
            Set parameters
</h5>,
          <SelectContentArea
            handleSelect={[Function]}
/>,
          <hr />,
          <Preview
            url="https://ephtracking.cdc.gov/apigateway/api/v1/indicators/null"
            validUrl={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retrieving List of all Content Areas",
            "className": "title",
          },
          "ref": null,
          "rendered": "Retrieving List of all Content Areas",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Usage",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Usage",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "https://ephtracking.cdc.gov/apigateway/api/",
              "{",
              "version",
              "}",
              "/indicators/",
              "{",
              "contentAreaId",
              "}",
              "[?apiToken]",
            ],
          },
          "ref": null,
          "rendered": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/indicators/",
            "{",
            "contentAreaId",
            "}",
            "[?apiToken]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Set parameters",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Set parameters",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSelect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/indicators/null",
            "validUrl": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
